Modules/_functoolsmodule.c

```diff
@@ -932,15 +932,31 @@ _functools_cmp_to_key_impl(PyObject *module, PyObject *mycmp)
 
 /* reduce (used to be a builtin) ********************************************/
 
-// Not converted to argument clinic, because of `args` in-place modification.
-// AC will affect performance.
+/*[clinic input]
+_functools.reduce
+
+    function as func: object
+    iterable as seq: object
+    initial as result: object(c_default="NULL") = _functools._initial_missing
```

Comment:
Do we really need to add this `_initial_missing` object?

Reply:
I see no other options, just as in the pure-Python version.  What you could suggest instead?  ``None`` and ``...`` were rejected.

If this is no-go - perhaps reduce() shouldn't be converted to AC, until it get support for multiple signatures.

Reply:
An alternative is to write the c_default as NULL and manually write an `@text_signature`. This is what is done by `_collections.deque.index`.

Reply:
Ah, I see, that's an option.  The drawback is that the inspect.signature() can't parse this and pure-Python version will have a different signature.

I'll postpone this change till @erlend-aasland review.

Reply:
Yes, my gut feel when I first browsed this PR earlier today, was also to use `@text_signature`. I think we should consider it. I'm not sure how relevant differences between the C extension module and the Python version of `functools` are.